packages/block-library/src/comments-pagination-numbers/index.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Server-side rendering of the `core/comments-pagination-numbers` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/comments-pagination-numbers` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the pagination numbers for the comments.
+ */
+function render_block_core_comments_pagination_numbers( $attributes, $content, $block ) {
+	// Get the post ID from which comments should be retrieved.
+	$post_id = isset( $block->context['postId'] )
+		? $block->context['postId']
+		: get_the_id();
+
+	if ( ! $post_id ) {
+		return '';
```

Comment:
It seems a bit limiting because it prevents using the pagination with a general list of comments like all the latest comments on the site. Maybe it's fine for start. Do we want to use pagination for comments that aren’t bound to a post id?

Reply:
Yep, I haven't thought of that case, to be honest ― but it is a good one. 🤔 

The [`get_approved_comments`](https://developer.wordpress.org/reference/functions/get_approved_comments/) function requires a `post_id`, but we can use [`get_comments`](https://developer.wordpress.org/reference/functions/get_comments/) instead and get all approved comments. I'll do that.

Reply:
Well, we would have to implement that behavior in the Comments Template block as well. Right now, it renders nothing if there is no `post_id`:

https://github.com/WordPress/gutenberg/blob/d51ac8a86b1756da64840b02f2ab5c129443391a/packages/block-library/src/comment-template/index.php#L22-L25

Maybe it's a better idea to implement this in a different PR. 🙂

Reply:
Sure, it's a general issue we have at the moment because we use the latest comments to visualize the Comments Query Loop block. It's something we have to sort out separately.